lib/rules/no-restricted-imports.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * @fileoverview Restrict usage of specified node imports.
+ * @author Guy Ellis
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+    var restrictedImports = context.options;
+
+    // if no imports are restricted we don"t need to check
+    if (restrictedImports.length === 0) {
+        return {};
+    }
+
+    return {
+        "ImportDeclaration": function(node) {
+            if (node && node.source && node.source.value) {
+
+                var value = node.source.value.trim();
+
+                if (restrictedImports.indexOf(value) !== -1) {
```

Comment:
This just checks for the presence of a substring. So if I say to block "os", it would also block " osx", "joust", etc. I think you want to just use `===` here.